node/mock-test.ts

```diff
@@ -224,58 +256,76 @@ export class MockTestRunner {
-        }
-
-        // Write marker to indicate download completed.
-        const marker = downloadDestination + '.completed';
-        fs.writeFileSync(marker, '');
+    private async downloadNode(nodeVersion: string, downloadDestination: string): Promise<string> {
+        return new Promise(async (resolve, reject) => {
+            shelljs.rm('-rf', downloadDestination);
+            let nodeUrl: string = process.env['TASK_NODE_URL'] || 'https://nodejs.org/dist';
+            nodeUrl = nodeUrl.replace(/\/$/, '');  // ensure there is no trailing slash on the base URL
+            let downloadPath = '';
+            switch (this.getPlatform()) {
+                case 'darwin':
+                    await this.downloadTarGz(nodeUrl + '/' + nodeVersion + '/node-' + nodeVersion + '-darwin-x64.tar.gz', downloadDestination);
+                    downloadPath = path.join(downloadDestination, 'node-' + nodeVersion + '-darwin-x64', 'bin', 'node');
+                    resolve(downloadPath)
+                    break;
+                case 'linux':
+                    await this.downloadTarGz(nodeUrl + '/' + nodeVersion + '/node-' + nodeVersion + '-linux-x64.tar.gz', downloadDestination);
+                    downloadPath = path.join(downloadDestination, 'node-' + nodeVersion + '-linux-x64', 'bin', 'node');
+                    resolve(downloadPath)
+                    break;
+                case 'win32':
+                    await this.downloadFile(nodeUrl + '/' + nodeVersion + '/win-x64/node.exe', downloadDestination, 'node.exe');
+                    await this.downloadFile(nodeUrl + '/' + nodeVersion + '/win-x64/node.lib', downloadDestination, 'node.lib');
+                    downloadPath = path.join(downloadDestination, 'node.exe')
+                    resolve(downloadPath)
+            }
 
-        return downloadPath;
+            // Write marker to indicate download completed.
+            const marker = downloadDestination + '.completed';
+            fs.writeFileSync(marker, '');
+            resolve(downloadPath);
+        })
     }
 
     // Downloads file to the downloadDestination, making any necessary folders along the way.
-    private downloadFile(url: string, downloadDestination: string, fileName: string): void {
-        const filePath: string = path.join(downloadDestination, fileName);
-        if (!url) {
-            throw new Error('Parameter "url" must be set.');
-        }
-        if (!downloadDestination) {
-            throw new Error('Parameter "downloadDestination" must be set.');
-        }
-        console.log('Downloading file:', url);
-        shelljs.mkdir('-p', downloadDestination);
-        const result: any = syncRequest('GET', url);
-        fs.writeFileSync(filePath, result.getBody());
+    private async downloadFile(url: string, downloadDestination: string, fileName: string): Promise<void> {
+        return await new Promise(async (resolve, reject) => {
```

Comment:
Nit: Can we get rid of `Promise` wrapper here? I believe `async/await` should be enough

Reply:
done